File: services/orders-service/src/main/java/pl/xsware/orders/application/event/PaymentFailedEvent.java
package pl.xsware.orders.application.event;

import java.time.Instant;
import java.util.UUID;

public record PaymentFailedEvent(
    UUID eventId,
    String eventType,
    int version,
    Instant occurredAt,
    Data data
) {

    public static final String TYPE = "PaymentFailed";
    public static final int VERSION = 1;

    public record Data(
        UUID orderId,
        UUID paymentId,
        String reason
    ) {}
}

File: services/payments-service/src/main/kotlin/pl/xsware/payments/application/event/PaymentFailedEvent.kt
package pl.xsware.payments.application.event

import java.time.Instant
import java.util.UUID

data class PaymentFailedEvent(
    val eventId: UUID,
    val eventType: String,
    val version: Int,
    val occurredAt: Instant,
    val data: Data
) {
    companion object {
        const val TYPE = "PaymentFailed"
        const val VERSION = 1
    }

    data class Data(
        val paymentId: UUID,
        val orderId: UUID,
        val reason: String? = null
    )
}
